create_jenkins_job.py

```diff
@@ -114,73 +116,89 @@ def main(argv=None):
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        create_job(os_name, 'ci_' + os_name, 'ci_job.xml.em', {
+            'cmake_build_type': 'None',
+        })
 
         # configure a manual version of the packaging job
-        job_name = 'ci_packaging_' + os_name
-        job_data['cmake_build_type'] = 'RelWithDebInfo'
-        job_data['test_bridge_default'] = 'true'
-        job_config = expand_template('packaging_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
-
-        # all following jobs are triggered nightly with email notification
-        job_data['time_trigger_spec'] = '30 7 * * *'
-        # for now, skip emailing about Windows failures
-        job_data['mailer_recipients'] = 'ros2-buildfarm@googlegroups.com'
+        create_job(os_name, 'ci_packaging_' + os_name, 'packaging_job.xml.em', {
+            'cmake_build_type': 'RelWithDebInfo',
+            'test_bridge_default': 'true',
+        })
 
         # configure packaging job
-        job_name = 'packaging_' + os_name
-        job_data['cmake_build_type'] = 'RelWithDebInfo'
-        job_data['test_bridge_default'] = 'true'
-        job_config = expand_template('packaging_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        create_job(os_name, 'packaging_' + os_name, 'packaging_job.xml.em', {
+            'cmake_build_type': 'RelWithDebInfo',
+            'test_bridge_default': 'true',
+            'time_trigger_spec': PERIODIC_JOB_SPEC,
+            'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+        })
 
-        # keeping the paths on Windows shorter
-        os_name = os_name.replace('windows', 'win')
         # configure nightly triggered job
-        job_name = 'nightly_' + os_name + '_debug'
-        if os_name == 'win':
-            job_name = job_name[0:-2]
-        job_data['cmake_build_type'] = 'Debug'
-        job_config = expand_template('ci_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        short_name = 'nightly_' + os_name + '_debug'
+        if os_name == 'windows':
+            short_name = 'nightly_win_deb'
+        create_job(os_name, short_name, 'ci_job.xml.em', {
+            'cmake_build_type': 'Debug',
+            'time_trigger_spec': PERIODIC_JOB_SPEC,
+            'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+        })
 
         # configure nightly coverage job on x86 Linux only
         if os_name == 'linux':
-            job_name = 'nightly_' + os_name + '_coverage'
-            job_data['cmake_build_type'] = 'Debug'
-            job_data['enable_c_coverage_default'] = 'true'
-            job_config = expand_template('ci_job.xml.em', job_data)
-            configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
-            job_data['enable_c_coverage_default'] = 'false'
+            create_job(os_name, 'nightly_linux_coverage', 'ci_job.xml.em', {
```

Comment:
Same remark asabove, I'm not sure hardcoding the job names is a good idea. If we want to create a coverage job for osx we will need to create another conditional bloc while we could just us this one if it stayed as it was before